kotlin-mbedtls/src/main/kotlin/org/opencoap/ssl/transport/DtlsServer.kt

```diff
@@ -64,7 +65,15 @@ class DtlsServer(
             dtlsState is DtlsSession -> dtlsState.decrypt(buf)
 
             // no session, but dtls packet contains CID
-            cid != null -> ReceiveResult.CidSessionMissing(cid!!)
+            cid != null -> {
+                if (sslConfig.cidSupplier != null && !sslConfig.cidSupplier.isValidCid(cid!!)) {
```

Comment:
`cid!!` is unnecessary because we are here only when `cid != null` as per the condition.

```suggestion
                if (sslConfig.cidSupplier?.isValidCid(cid) == false) {
```